src/os/style/label.js

```diff
@@ -505,41 +493,134 @@ os.style.label.createAdditionalLabels = function(feature, config, opt_layerConfi
+    if (typeof fontSize == 'string') {
+      fontSize = parseInt(fontSize, 10) || undefined;
+    }
+
+    fontSize = fontSize || os.style.label.DEFAULT_SIZE;
+
+    const labelFont = os.style.label.getFont(fontSize);
+    textStyle.setFont(labelFont);
   }
+};
 
-  if (!fontSize) {
-    fontSize = os.style.label.DEFAULT_SIZE;
+/**
+ * @param {ol.style.Style} labelStyle
+ * @param {ol.Feature} feature
+ * @param {Object} config
+ * @param {Object=} opt_layerConfig
+ */
+os.style.label.updateDefaultFillColor = function(labelStyle, feature, config, opt_layerConfig) {
+  if (!config || !config['text'] || (
+    config['text']['fillColor'] === undefined && (
+      config['text']['fill'] === undefined ||
+          config['text']['fill']['color'] === undefined))) {
+    const textStyle = labelStyle.getText();
+    const labelColor = os.style.label.getColor(feature, config, opt_layerConfig);
+    const fill = textStyle.getFill();
+    if (fill) {
+      fill.setColor(labelColor);
+    }
   }
+};
 
-  var labelFont = os.style.label.getFont(fontSize);
-  textStyle.setFont(labelFont);
+/**
+ * @param {ol.style.Style} labelStyle
+ * @param {ol.Feature} feature
+ * @param {Object} config
+ * @param {Object=} opt_layerConfig
+ */
+os.style.label.updateDefaultStrokeColor = function(labelStyle, feature, config, opt_layerConfig) {
+  if (!config || !config['text'] || (
+    config['text']['strokeColor'] === undefined && (
+      config['text']['stroke'] === undefined ||
+      config['text']['stroke']['color'] === undefined))) {
+    const textStyle = labelStyle.getText();
+    const labelColor = os.style.label.getColor(feature, config, opt_layerConfig);
+    const stroke = textStyle.getStroke();
+    if (stroke) {
+      const fillColor = ol.color.asArray(labelColor);
+      const strokeColor = ol.color.asArray(/** @type {Array<number>|string} */ (stroke.getColor()));
+      strokeColor[3] = fillColor[3];
+      stroke.setColor(os.style.toRgbaString(strokeColor));
+    }
+  }
+};
 
-  var labelColor = os.style.label.getColor(feature, config, opt_layerConfig);
+/**
+ * @param {ol.style.Style} labelStyle
+ * @param {string} label
+ */
+os.style.label.updateText = function(labelStyle, label) {
+  const text = labelStyle.getText();
+  text.setText(os.style.label.prepareText(label, true));
+};
 
-  var fill = textStyle.getFill();
-  if (fill) {
-    fill.setColor(labelColor);
+/**
+ * @param {ol.style.Style} labelStyle
+ * @param {Object} config
+ */
+os.style.label.updateDefaultOffsetX = function(labelStyle, config) {
+  // labels need to be offset a little more when next to an icon. this helps, but isn't nearly complete.
+  // TODO: determine the size of the rendered feature and use that for the x offset
+  if (!config || !config['text'] || config['text']['offsetX'] === undefined) {
+    const text = labelStyle.getText();
+    const fontSize = os.style.label.DEFAULT_SIZE;
+    const offsetx = os.style.isIconConfig(config) ? fontSize : (fontSize / 2);
```

Comment:
The `baseLabelConfig` passed into this function does not have information from the original config that `isIconConfig` depends on. This check always returns false, which is changing the default x-offset for icons.

Reply:
Should be fixed in my recent update.